tests/x11/nis_client_validate.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate YaST configuration functionality for NIS
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use scheduler 'get_test_suite_data';
+use lockapi 'mutex_lock';
+use x11utils 'turn_off_gnome_screensaver';
+use utils 'systemctl';
+
+sub run {
+    my ($self) = @_;
+    my $test_data = get_test_suite_data();
+
+    x11_start_program('xterm -geometry 155x45+5+5', target_match => 'xterm');
+    turn_off_gnome_screensaver if check_var('DESKTOP', 'gnome');
+    become_root;
+
+    assert_script_run "ls /etc/yp.conf",              fail_message => "File /etc/yp.conf doesn't exist";
+    assert_script_run "grep \"ypserv\" /etc/yp.conf", fail_message => "\"ypserv\" was not found in /etc/yp.conf";
+    assert_script_run "grep \"$test_data->{nis_domain}\" /etc/defaultdomain",
+      fail_message => "\"$test_data->{nis_domain}\" was not found in /etc/defaultdomain";
+    mutex_lock('nis_user_ready');
```

Comment:
Looks like we are missing mutex_unlock('nis_user_ready'); In case we don't have any critical code section, we can simply replace it with `mutex_wait` which does lock and unlock.

Reply:
As discussed, changing the mutex doesn't work.

Reply:
Seems that we have some misunderstanding here. Currently we have only lock mutex and never unlock, so it wait's till mutex is created, and then we use `wait_for_children` which doesn't rely on mutexes at all and just wait till job is done. So to be 100% right and not to face issues in future if we decide to extend this part, we should release the lock at some point. That's why I've proposed `mutex_wait` because it will wait till lock is created to lock it and will release it immediately, as we don't have any critical section here. If we do, let's call `mutex_unlock` explicitly in the end of the critical section.

Reply:
Done